### PR TITLE
Websockets rebase to master

### PR DESCRIPTION
**Pull request by erenatas**

### CODE REVIEW

File: binance_trade_bot/binance_api_manager.py [hunk 1]
```diff
@@ -136,15 +138,73 @@ def wait_for_order(self, order_id) -> BinanceOrder:
 
         self.logger.debug(f"Order created: {order_status}")
 
-        while order_status.status != "FILLED":
-            order_status = self.cache.orders.get(order_id, None)
-            self.logger.debug(f"Waiting for order {order_id} to be filled")
-            time.sleep(1)
+        while order_status["status"] != "FILLED":
```
Comment (j-waters) [inline binance_trade_bot/binance_api_manager.py]: I don't think this is quite right, `order_status` should be fetched from the cache, and will be of type `BinanceOrder`, not a dictionary

**Review by j-waters — CHANGES_REQUESTED** [review]
I think it looks okay other than this, I haven't tried to run it though

I remember you saying you tested it using backtesting, have you tested it normally?

File: binance_trade_bot/binance_api_manager.py (hunk 1 above)
Comment (erenatas) [reply binance_trade_bot/binance_api_manager.py]: Thanks for pointing out. Debugging at the moment. Fixed this part. Trying to understand why pylint keeps skipping locally on my side...

**Comment by GrigorovskyA** [comment]
I met  next problem with that branch
```PS D:\share\binance-trade-websock_upd> python -m binance_trade_bot
2021-04-27 19:58:50,135 - crypto_trading_logger - INFO - Starting
2021-04-27 19:58:50,807 - crypto_trading_logger - INFO - Chosen strategy: multiple_coins
2021-04-27 19:58:50,807 - crypto_trading_logger - INFO - Creating database schema if it doesn't already exist
2021-04-27 19:58:59,156 - crypto_trading_logger - ERROR - Error while scouting...
Traceback (most recent call last):
  File "D:\share\binance-trade-websock_upd\binance_trade_bot\scheduler.py", line 25, in _run_job
    super()._run_job(job)
  File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 147, in _run_job
    ret = job.run()
  File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 491, in run
    ret = self.job_func()
  File "D:\share\binance-trade-websock_upd\binance_trade_bot\strategies\multiple_coins_strategy.py", line 39, in scout
    f"{datetime.now()} - CONSOLE - INFO - I am scouting the best trades. "
TypeError: unsupported operand type(s) for +: 'NoneType' and 'Coin'

2021-04-27 19:59:02,055 - crypto_trading_logger - ERROR - Error while scouting...
Traceback (most recent call last):
  File "D:\share\binance-trade-websock_upd\binance_trade_bot\scheduler.py", line 25, in _run_job
    super()._run_job(job)
  File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 147, in _run_job
    ret = job.run()
  File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 491, in run
    ret = self.job_func()
  File "D:\share\binance-trade-websock_upd\binance_trade_bot\strategies\multiple_coins_strategy.py", line 39, in scout
    f"{datetime.now()} - CONSOLE - INFO - I am scouting the best trades. "
TypeError: unsupported operand type(s) for +: 'NoneType' and 'Coin'
```

File: binance_trade_bot/binance_api_manager.py (hunk 1 above)
Comment (erenatas) [reply binance_trade_bot/binance_api_manager.py]: I updated the change. Ran normally too. Can you test?

**Comment by erenatas** [comment]
> I met next problem with that branch
> 
> ```postscript
> 2021-04-27 19:58:50,135 - crypto_trading_logger - INFO - Starting
> 2021-04-27 19:58:50,807 - crypto_trading_logger - INFO - Chosen strategy: multiple_coins
> 2021-04-27 19:58:50,807 - crypto_trading_logger - INFO - Creating database schema if it doesn't already exist
> 2021-04-27 19:58:59,156 - crypto_trading_logger - ERROR - Error while scouting...
> Traceback (most recent call last):
>   File "D:\share\binance-trade-websock_upd\binance_trade_bot\scheduler.py", line 25, in _run_job
>     super()._run_job(job)
>   File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 147, in _run_job
>     ret = job.run()
>   File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 491, in run
>     ret = self.job_func()
>   File "D:\share\binance-trade-websock_upd\binance_trade_bot\strategies\multiple_coins_strategy.py", line 39, in scout
>     f"{datetime.now()} - CONSOLE - INFO - I am scouting the best trades. "
> TypeError: unsupported operand type(s) for +: 'NoneType' and 'Coin'
> 
> 2021-04-27 19:59:02,055 - crypto_trading_logger - ERROR - Error while scouting...
> Traceback (most recent call last):
>   File "D:\share\binance-trade-websock_upd\binance_trade_bot\scheduler.py", line 25, in _run_job
>     super()._run_job(job)
>   File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 147, in _run_job
>     ret = job.run()
>   File "C:\Users\Elmin\AppData\Local\Programs\Python\Python38-32\lib\site-packages\schedule\__init__.py", line 491, in run
>     ret = self.job_func()
>   File "D:\share\binance-trade-websock_upd\binance_trade_bot\strategies\multiple_coins_strategy.py", line 39, in scout
>     f"{datetime.now()} - CONSOLE - INFO - I am scouting the best trades. "
> TypeError: unsupported operand type(s) for +: 'NoneType' and 'Coin'
> ```

Updated, can you try again?

**Comment by GrigorovskyA** [comment]
> Updated, can you try again?

Yeah, and I forget to update branch to websocket after cloning =/
Now bot started fine! Keep testing

**Comment by erenatas** [comment]
Fixed the pylint issue. Sorry for not seeing...

**Pull request closed without merging**